### DISCUSSION TITLE
Query condition base on the next node

### DISCUSSION THREAD

**Discussion by maxxnino** [nvim-treesitter/nvim-treesitter#1658] [Q&A]
When writing a query I have two scenario:
 - Scenario 1: if a **(SuffixOp)** node don't have **(FnCallArguments)** node below it. The  **(IDENTIFIER)** node become a field
 -  Scenario 2: if a **(SuffixOp)** node have, The  **(IDENTIFIER)** node become a function

How can i achive that?

```scheme
(ErrorUnionExpr
  ; Scenario 1
  (SuffixOp (IDENTIFIER) @field)

  ; Scenario 2
  (SuffixOp (IDENTIFIER) @function)
  (FnCallArguments)

  ; Scenario 1
  (SuffixOp (IDENTIFIER) @field)

  ; Scenario 2
  (SuffixOp (IDENTIFIER) @function)
  (FnCallArguments)
)
```

**Comment by maxxnino** [comment]
I re-worked on grammar, make it easier for query

**Reply by vigoux** [reply]
Is the issue still here ? I'll try to find an answer if yes.

**Reply by maxxnino** [reply]
I solved the issue with [anchor operator](https://tree-sitter.github.io/tree-sitter/using-parsers#query-syntax). And it's already in the zig query https://github.com/nvim-treesitter/nvim-treesitter/blob/460a26ef3218057a544b3fd6697e979e6bad648d/queries/zig/highlights.scm#L76-L91
